spec/javascripts/govuk_publishing_components/analytics-ga4/gtm-click-tracking.spec.js

```diff
@@ -0,0 +1,94 @@
+/* eslint-env jasmine */
+
+describe('Google Tag Manager click tracking', function () {
+  var GOVUK = window.GOVUK
+  var element
+
+  beforeEach(function () {
+    window.dataLayer = []
+    element = document.createElement('div')
+  })
+
+  afterEach(function () {
+    document.body.removeChild(element)
+  })
+
+  describe('doing simple tracking on a single element', function () {
+    beforeEach(function () {
+      element.setAttribute('data-gtm-event-name', 'event-name')
+      var attributes = {
+        'test-1': 'test-1 value',
+        'test-2': 'test-2 value'
+      }
+      element.setAttribute('data-gtm-attributes', JSON.stringify(attributes))
+      document.body.appendChild(element)
+      new GOVUK.Modules.GtmClickTracking(element).init()
+    })
+
+    it('pushes gtm attributes to the dataLayer', function () {
+      element.click()
+      var expected = {
+        event: 'analytics',
+        event_name: 'event-name',
+        link_url: window.location.href.substring(window.location.origin.length),
+        ui: {
+          'test-1': 'test-1 value',
+          'test-2': 'test-2 value'
+        }
+      }
+      expect(window.dataLayer[0]).toEqual(expected)
+    })
+  })
+
+  describe('doing simple tracking on multiple elements', function () {
+    beforeEach(function () {
+      var attributes1 = {
+        'test-1-1': 'test 1-1 value',
+        'test-1-2': 'test 1-2 value'
+      }
+      var attributes2 = {
+        'test-2-1': 'test 2-1 value',
+        'test-2-2': 'test 2-2 value'
+      }
+      element.innerHTML =
+        '<div data-gtm-event-name="event1-name"' +
+          'data-gtm-attributes=\'' + JSON.stringify(attributes1) + '\'' +
```

Comment:
To avoid the need for escaping and to be consistent with the previous and next line, double quotes could be used:
```suggestion
          'data-gtm-attributes="' + JSON.stringify(attributes1) + '"' +
```

Reply:
I thought so too - but it doesn't work.

`stringify` returns a string containing `"` so the rendered HTML becomes something invalid like `<div data-gtm-attributes="{ "attribute": "value" }">` and the test fails.

(I was going to switch the quotes/ticks around but the linter enforces `'` on the outer boundary)